src/session/content/chat_action_bar.rs

```diff
@@ -165,6 +175,63 @@ mod imp {
                 .connect_activate(clone!(@weak obj => move |_| {
                     obj.activate_action("chat-action-bar.send-text-message", None).unwrap()
                 }));
+
+            // Mute/unmute channel
+            self.mute_button
+                .connect_clicked(clone!(@weak obj => move |_| {
```

Comment:
Please use actions instead of signals to react to button press. You can look around for how it's done in other places.